torch/nn/parallel/distributed.py

```diff
@@ -296,7 +296,7 @@ def _process_batch():
             # Wait for all copies to complete before starting the NCCL kernel
             for stream in reduction_streams:
                 stream.synchronize()
-            nccl.reduce(dev_coalesced, root=device_ids[0], streams=nccl_streams)
+            nccl.reduce(dev_coalesced, root=0, streams=nccl_streams)
```

Comment:
0 Is not the device id, it's the rank relative to the nccl communicator. We could likely skip some sync params, but will still have to sync buffers.